packages/react-native-sdk/src/components/Livestream/LivestreamPlayer/LivestreamPlayer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo, useEffect } from 'react';
+import {
+  ViewerLivestream as DefaultViewerLivestream,
+  ViewerLivestreamProps,
+} from '..';
+import { StreamCall, useStreamVideoClient } from '../../..';
+
+export type LivestreamPlayerProps = {
+  /**
+   * The call type. Usually `livestream`.
+   */
+  callType: string;
+  /**
+   * The call ID.
+   */
+  callId: string;
+  /**
+   * Component to override the ViewerLivestream component used under the hood.
+   * **Default** [ViewerLivestream](https://github.com/GetStream/stream-video-js/blob/main/packages/react-native-sdk/src/components/Livestream/ViewerLivestream/ViewerLivestream.tsx)
+   */
+  ViewerLivestream?: React.ComponentType<ViewerLivestreamProps>;
+};
+
+export const LivestreamPlayer = ({
+  callType,
+  callId,
+  ViewerLivestream = DefaultViewerLivestream,
+}: LivestreamPlayerProps) => {
+  const client = useStreamVideoClient();
+
+  const call = useMemo(
+    () => client?.call(callType, callId),
+    [callType, callId, client],
+  );
+
+  useEffect(() => {
+    if (!call) {
+      return;
+    }
+    call.join().catch((e) => {
```

Comment:
Can we have try catch instead of .then.catch?

Reply:
my opinion is that it doesnt matter here. Both are fine.

When we chain promises I would rather go for try/catch, for single promise, just having catch is fine in a useEffect scenario